app/cameratakelib/src/main/java/com/genar/hmssandbox/huawei/cameratakelib/utils/FileUtil.java

```diff
@@ -0,0 +1,334 @@
+package com.genar.hmssandbox.huawei.cameratakelib.utils;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.ImageFormat;
+import android.graphics.Matrix;
+import android.graphics.Rect;
+import android.media.Image;
+import android.opengl.GLES20;
+import android.os.Environment;
+import android.os.StatFs;
+import android.text.TextUtils;
+
+import com.genar.hmssandbox.huawei.modelingresource.util.Constants;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.IntBuffer;
+
+import top.zibin.luban.CompressionPredicate;
+import top.zibin.luban.Luban;
+import top.zibin.luban.OnCompressListener;
+
+public class FileUtil {
+
+    /**
+     * 计算Sdcard的剩余大小
+     *
+     * @return MB
+     */
+    public static long getAvailableSize() {
+        //sd卡大小相关变量
+        StatFs statFs;
+        File file = Environment.getExternalStorageDirectory();
+        statFs = new StatFs(file.getPath());
+        //获得Sdcard上每个block的size
+        long blockSize = statFs.getBlockSize();
+        //获取可供程序使用的Block数量
+        long blockavailable = statFs.getAvailableBlocks();
+        //计算标准大小使用：1024，当然使用1000也可以
+        long blockavailableTotal = blockSize * blockavailable / 1024 / 1024;
+        return blockavailableTotal;
+    }
+
+    /**
+     * SDCard 总容量大小
+     *
+     * @return MB
+     */
+    public static long getTotalSize() {
+        StatFs statFs;
+        File file = Environment.getExternalStorageDirectory();
+        statFs = new StatFs(file.getPath());
+        //获得sdcard上 block的总数
+        long blockCount = statFs.getBlockCount();
+        //获得sdcard上每个block 的大小
+        long blockSize = statFs.getBlockSize();
+        //计算标准大小使用：1024，当然使用1000也可以
+        long bookTotalSize = blockCount * blockSize / 1024 / 1024;
+        return bookTotalSize;
+    }
+
+    /**
+     * 保存bitmap到本地
+     *
+     * @param bitmap
+     * @return
+     */
+    public static File saveBitmap(Context context, Bitmap bitmap, String createTime, Integer index) {
+        String savePath = new Constants(context).getCaptureMaterialImageFile() + createTime + "/";
+//        String savePath = "/data/data/com.huawei.hms.modeling/files/3DMagic" + createTime + "/";
+//        String savePath = "/sdcard/3DMagic/3DMagic" + createTime + "/";
+        File filePic;
+        try {
+            filePic = new File(savePath + index + ".jpg");
+            if (!filePic.exists()) {
+                if (!filePic.getParentFile().exists()) {
+                    filePic.getParentFile().mkdirs();
+                }
+                filePic.createNewFile();
+            }
+            FileOutputStream fos = new FileOutputStream(filePic);
+            bitmap.compress(Bitmap.CompressFormat.JPEG, 80, fos);
+            fos.flush();
```

Comment:
all streams have to close in the finally scope or after the catch: ( **fos.close()** )